nncf/experimental/torch/fx/transformations.py

```diff
@@ -529,9 +582,206 @@ def fuse_conv_bn(model: torch.fx.GraphModule) -> None:
+        if quantize_node.target != QDQ_PAIR[node.target]:
+            continue
+        port_id = 0
+        args = _get_node_inputs(quantize_node, model)
+        if not args:
+            continue
+        if quantize_node.target == torch.ops.quantized_decomposed.quantize_per_channel.default:
+            _reshape_scale_zp(model, quantize_node)
+        result = quantize_node.target(*args)
+        constant_update_fn(
+            model, quantize_node, result, port_id, updated_node_name="compressed_weight_updated_constant"
+        )
+
+
+def _reshape_scale_zp(model: torch.fx.GraphModule, node: torch.fx.Node) -> None:
+    """
+    Reshape scale and zero point so that it can be multiplied elementwise with the weight for per channel quantization.
+
+    :param model: Model to apply transformations to.
+    :param node: quantize node whose scale has to be reshaped
+    """
+    weight_node = node.all_input_nodes[0]
+    scale_node = node.all_input_nodes[1]
+    zp_node = node.all_input_nodes[2]
+    axis = node.args[3]
+    scale_value = get_tensor_constant_from_node(scale_node, model)
+    weight_value = get_tensor_constant_from_node(weight_node, model)
+    zp_value = get_tensor_constant_from_node(zp_node, model)
+    new_shape = [1] * weight_value.dim()
+    new_shape[axis] = scale_value.shape[0]
+    scale_value = scale_value.reshape(new_shape)
+    zp_value = zp_value.reshape(new_shape)
+    constant_update_fn(model, node, zp_value, 2, updated_node_name=zp_node.name + "_updated_constant")
+    constant_update_fn(model, node, scale_value, 1, updated_node_name=scale_node.name + "_updated_constant")
+
+
+def fq_weights_transformation(model: torch.fx.GraphModule) -> None:
+    """
+    This function applies a transformation to replace the FP32 weights with Fake Quantized
+    FP values to avoid the rounding error when converting to OpenVino model.
+
+    :param model: Model to apply transformations to.
+    """
+    for node in model.graph.nodes:
+        if node.target not in DEQUANTIZE_NODE_TARGETS:
+            continue
+        port_id = 0
+        quantize_node = node.all_input_nodes[0]
+        # Case where Quantize Dequantize nodes are not consecutive
+        if quantize_node.target != QDQ_PAIR[node.target]:
+            continue
+        args = _get_node_inputs(quantize_node, model)
+        if not args:
+            continue
+        result = node.target(quantize_node.target(*args), *args[1:])
+        constant_update_fn(
+            model,
+            quantize_node,
+            result,
+            port_id,
+            updated_node_name="compressed_weight_updated_constant",
+        )
+
+
+def compress_post_quantize_transformation(model: torch.fx.GraphModule) -> None:
+    """
+    Applies transformation to compress the weights to Int8 after the quantization step.
+
+    :param model: Model to apply transformations to.
+    """
+    _compress_qdq_constant_transformation(model)
+    _remove_constant_qdq_transformation(model)
```

Comment:
It looks like `_compress_qdq_constant_transformation` returns incorrect model. I mean the model could not be infered. I suggest to simplify structure of this function and do all changes in one transformation which replace the FP32->q->dq pattern to the INT8-> sub -> mul pattern. Inside of the pattern you can calculate INT8 constant and reshape scale and zero_point using util functions.

Reply:
Alright, this seems more robust. I will do it 👍

Reply:
Done

Reply:
Could you please move the all code from `_remove_constant_qdq_transformation` to `compress_post_quantize_transformation`?